test/builder/QuarkBuilderMorphoBorrow.t.sol

```diff
@@ -0,0 +1,645 @@
+        assertEq(result.actions.length, 2, "two actions");
+        // first action
+        assertEq(result.actions[0].chainId, 1, "operation is on chainid 1");
+        assertEq(result.actions[0].quarkAccount, address(0xa11ce), "0xa11ce sends the funds");
+        assertEq(result.actions[0].actionType, "BRIDGE", "action type is 'BRIDGE'");
+        assertEq(result.actions[0].paymentMethod, "PAY_CALL", "payment method is 'PAY_CALL'");
+        assertEq(result.actions[0].paymentToken, USDC_1, "payment token is USDC on mainnet");
+        assertEq(result.actions[0].paymentMaxCost, 0.1e6, "payment should have max cost of 0.1e6");
+        assertEq(
+            result.actions[0].actionContext,
+            abi.encode(
+                Actions.BridgeActionContext({
+                    amount: 1e6,
+                    price: USDC_PRICE,
+                    token: USDC_1,
+                    assetSymbol: "USDC",
+                    chainId: 1,
+                    recipient: address(0xa11ce),
+                    destinationChainId: 8453,
+                    bridgeType: Actions.BRIDGE_TYPE_CCTP
+                })
+            ),
+            "action context encoded from BridgeActionContext"
+        );
+        // second action
+        assertEq(result.actions[1].chainId, 8453, "operation is on chainid 8453");
+        assertEq(result.actions[1].quarkAccount, address(0xa11ce), "0xa11ce sends the funds");
+        assertEq(result.actions[1].actionType, "MORPHO_BORROW", "action type is 'MORPHO_BORROW'");
+        assertEq(result.actions[1].paymentMethod, "PAY_CALL", "payment method is 'PAY_CALL'");
+        assertEq(result.actions[1].paymentToken, USDC_8453, "payment token is USDC on Base");
+        assertEq(result.actions[1].paymentMaxCost, 1e6, "payment should have max cost of 1e6");
+
+        assertEq(
+            result.actions[1].actionContext,
+            abi.encode(
+                Actions.MorphoBorrowActionContext({
+                    amount: 0.2e18,
+                    assetSymbol: "WETH",
+                    chainId: 8453,
+                    collateralAmount: 1e18,
+                    collateralTokenPrice: CBETH_PRICE,
+                    collateralToken: cbEth_(8453),
+                    collateralAssetSymbol: "cbETH",
+                    price: WETH_PRICE,
+                    token: weth_(8453),
+                    morpho: MorphoInfo.getMorphoAddress(8453),
+                    morphoMarketId: MorphoInfo.marketId(MorphoInfo.getMarketParams(8453, "cbETH", "WETH"))
+                })
+            ),
+            "action context encoded from MorphoBorrowActionContext"
+        );
+
+        // TODO: Check the contents of the EIP712 data
+        assertNotEq(result.eip712Data.digest, hex"", "non-empty digest");
+        assertNotEq(result.eip712Data.domainSeparator, hex"", "non-empty domain separator");
+        assertNotEq(result.eip712Data.hashStruct, hex"", "non-empty hashStruct");
+    }
```

Comment:
Will add `testBorrowMaxCostToohigh` case. 

For bridging collateral, ideally it should work out of box, but unfortunately with our current bridge routes only supports "usdc" (Since we only support CCTP so far) and so far we don't have borrow markets that allow "usdc" to be collateral. 
So I think that's the reason the bridging collateral scenarios is missing from both Comet and Morpho.